.github/workflows/release.main.kts

```diff
@@ -93,7 +94,8 @@ workflow(
         uses(
             name = "Upload to Codecov.io",
             action = CodecovAction(
-                failCiIfError = true
+                failCiIfError = true,
+                token = expr(CODECOV_TOKEN)
```

Comment:
Why is the token necessary for release but not for branches-and-prs?

Reply:
I think indeed you need it since some change on Codecov side that required using tokens on the main repo branches or something like that.

Reply:
Or alternatively you could since v5 also enable token-less uploads on codecov side for public repositories: https://docs.codecov.com/docs/codecov-tokens#uploading-without-a-token

Reply:
Yep, like I remembered.
The upload to Codecov was broken since using v4 of the action.
It only worked for PR builds from forked repositories.
For the renovate PRs or my GT-sent PRs Codecov was not working.
Just now with the Codecov improvements PR merged you now see it as it enabled CI failure on upload failure due to exactly such reasons.

So either the token is also needed for the other workflow, or - as we are on v5 - you need to enable tokenless upload on codecov side.

Reply:
https://docs.codecov.com/docs/codecov-tokens#uploading-without-a-token

Reply:
https://docs.codecov.com/docs/codecov-tokens#uploading-without-a-token
> the repository is public and is not on a "protected" branch

It should only require a token for the `master` branch

Reply:
Anyway, I've enabled tokenless upload for the organization.

Reply:
> It should only require a token for the master branch

Protected branch in CodeCov sense, not in GitHub sense.
From the same page:

> **Tokenless on Unprotected Branches**
>
> A protected branch corresponds to a permanent branch in your repository (like main or master). An unprotected branch is any branch with a colon-separated prefix on it (like forkname:main or pr300:master).
>
> Projects that accept contributions from forks can add a prefix to the contribution's branch name and send uploads to Codecov without having to expose any CI secrets.
>
> Starting on >v4.0 of the codecov-action, the branch name on pull requests from forks is automatically modified to include a :. Codecov should work smoothly on pull requests from forks out of the box.

So in the CodeCov sense you would need to prefix the branch name with something with colon to "mark" it as unprotected. The GHA action does exactly that for PR builds from forks, which is the reason it worked fine for PR builds from forks, but not for any branch in the main repository.